lab11.md

```diff
@@ -0,0 +1,173 @@
+# Lab 11 - Advanced feature - Trust with HMAC
+
+<img src="https://github.com/openfaas/media/raw/master/OpenFaaS_Magnet_3_1_png.png" width="500px"></img>
+
+Before starting this lab create new folder
+```
+mkdir -p lab11 \
+   && cd lab11
+```
+
+## What is HMAC
+
+Currently our functions can get information from everywhere without any restriction. If we want to validate the source of information we can use _Hash-based Message Authentication Code (HMAC)_.
+
+From [alexellis/hmac](https://github.com/alexellis/hmac):
+> HMAC uses a symmetric key that both sender/receiver share ahead of time. The sender will generate a hash when wanting to transmit a message - this data is sent along with the payload. The recipient will then sign payload with the shared key and if the hash matches then the payload is assumed to be from the sender.
+
+This way we close our functions from being invoked with invalid or even dangerous information.
+
+## Using HMAC
+
+For the purpose we will create new python3 function and let's give it a name `hmac-protected`:
```

Comment:
I’d suggest some rewording, e.g
“For our purpose  we are going to create a new Phython 3 function. Let’s call it...”